### DISCUSSION TITLE
Nightly build and old script mods

### DISCUSSION THREAD

**Discussion by MK77P** [scripthookvdotnet/scripthookvdotnet#1456] [Q&A] [answered]
So as I understand with the latest patch we have to use the nightly build of SHVDN and with that some of the scripts aren't working at all or work partially, which makes me wonder and ask...will there be an update in the future for SHVDN to fully work with the latest version of GTAV where all the scripts will be fulling working? or should I just roll back to the previous game patch and use SHVDN 3.6.0 if I want all of the scripts to work?

**Comment by kagikn** [comment]
You could roll back the game version and revert your SHVDN to v3.6.0 if you want it to run much more stably without having to wait.

**Reply by MK77P** [reply]
So SHVDN will eventually fully support the latest game patch?

**Reply by kagikn** [reply]
I don't think SHVDN has issues that happen in *everyone*'s game in the latest game patch no more. I think you're sort of unlucky if you getting game crash or infinite loading screen with a nightly SHVDN such as nightly.95.

**Reply by kagikn** [reply]
I'm fixing more of flaky issues btw, so more people can stably run SHVDN

**Reply by MK77P** [reply]
The issues that I'm having with the nightly build is that some scripts with the nightly build aren't working at all, is that supposed to happen?

**Reply by kagikn** [reply]
If you're saying some scripts built against nightly builds aren't working (for missing methods or something like that), blame the script authors. I never guarantee that nightly updates won't do breaking changes to anything of the API that aren't available in any stable versions (and crosire never did). The readme says the same thing.

**Reply by MK77P** [reply]
This [script](https://www.gta5-mods.com/scripts/iv-style-exit-vehicle-net) for example works perfectly fine if I was to use SHVDN 3.6.0, but if I was to use SHVDN nightly it doesn't work at all.

**Reply by kagikn** [reply, marked as answer]
Well in that case, did you add `.2` to the script file name right before the extension name `.cs`? You need to add it so the script continues to run under the v2 API, which is the same behavior as how SHVDN runs raw scripts without a version notation in v3.6.0.

If you still have some issues even after adding the version notation, let me know more, such as error logs or how the game behaves.

**Reply by MK77P** [reply]
Thank you! that did it, the script is now working, I did read that but didn't pay attention to it thinking it was for developers, my apologies.

And one last question, adding the ".2" should be done for every single .cs script that I'm using correct?

**Reply by kagikn** [reply]
Yes, adding the ".2" should be fine for most .cs scripts. I'm not aware of any scripts that are supposed to run under the v3 API but the `.3` is missing (making them being able to run in nightly versions without renaming). My apologies for not logging warnings for missing version notations to both the log file and the console, too.

**Reply by lucienlmy** [reply]
How to name a script with a .dll extension

**Reply by lucienlmy** [reply]
> Yes, adding the ".2" should be fine for most .cs scripts. I'm not aware of any scripts that are supposed to run under the v3 API but the `.3` is missing (making them being able to run in nightly versions without renaming). My apologies for not logging warnings for missing version notations to both the log file and the console, too.

How to name a script with a .dll extension

**Reply by kagikn** [reply]
No need to add version notations to file names of the .dll scripts like `Script.2.dll`, as they specify the API version to use within them.

**Reply by JohnFromGWN** [reply]
> Well in that case, did you add `.2` to the script file name right before the extension name `.cs`? You need to add it so the script continues to run under the v2 API, which is the same behavior as how SHVDN runs raw scripts without a version notation in v3.6.0.
> 
> If you still have some issues even after adding the version notation, let me know more, such as error logs or how the game behaves.

Really surprised by your comment. I'm editing an old script I wrote in SHVDN2 and running it as we speak as .cs without any issues. The version of SHVDN2 is 2.10.13.0 (the one with 3.6) so I can vouch that, at least with V2372, it does indeed work as .cs.
On that note, once the script is finished, I will convert it to DN3.
